service/service.go

```diff
@@ -283,6 +258,21 @@ func (s *Service) configuration() *Dependency {
 	return nil
 }
 
+func serviceHash(contextDir string, env map[string]string) ([]byte, error) {
+	// create stream to calculate hash
+	stream, err := archive.Tar(contextDir, archive.Uncompressed)
```

Comment:
Is this really necessary ? We already have the tarball in the `New` we could simply 
```go
        h := sha256.New()
	if _, err := io.Copy(h, tarball); err != nil {
		return nil, err
	}
	envbytes := []byte(xos.EnvMapToString(env))
	s.Hash := hex.EncodeToString(h.Sum(envbytes))
```

in the `New` with no need of creating another tar